.vib/metallb/ginkgo/metallb_test.go

```diff
@@ -0,0 +1,74 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MetalLB", func() {
+	var typedClient corev1.CoreV1Interface
+	var dynamicClient dynamic.Interface
+	var ctx context.Context
+
+	BeforeEach(func() {
+		typedClient = corev1.NewForConfigOrDie(clusterConfigOrDie())
+		dynamicClient = dynamic.NewForConfigOrDie(clusterConfigOrDie())
+		ctx = context.Background()
+	})
+
+	Context("configured via IPAddressPool & L2Advertisement resources (CRDs)", func() {
+		var ipPoolConfig = IpAddressPoolConfig{
+			name:        "ip-address-pool-vib-tests",
+			addressFrom: "192.168.100.0",
+			addressTo:   "192.168.100.255",
+		}
+		var advertConfig = L2AdvertisementConfig{
+			name:              "l2-advertisement-vib-tests",
+			ipAddressPoolName: ipPoolConfig.name,
+		}
+
+		BeforeEach(func() {
+			createIpAddressPoolResourceOrDie(ctx, dynamicClient, &ipPoolConfig)
+			createL2AdvertisementResourceOrDie(ctx, dynamicClient, &advertConfig)
+		})
+		AfterEach(func() {
+			// Not need to panic here if failed, the cluster is expected to clean up with the undeployment
+			dynamicClient.Resource(l2AdvertisementType).Namespace(*namespace).Delete(ctx, advertConfig.name, metav1.DeleteOptions{})
+			dynamicClient.Resource(ipAddressPoolType).Namespace(*namespace).Delete(ctx, ipPoolConfig.name, metav1.DeleteOptions{})
+		})
+
+		When("a Service resource is created", func() {
```

Comment:
Could we create the testing service at the chart's installation time instead? Checking Metallb doc, it should remain idle until configured with an IPAdressPool and L2Advertisement. In such a case, you may be able to create the LB using the `extraDeploys` param instead of adding this logic here. The testing logic will be the same except for the service creation.

Reply:
Although your proposed workflow makes sense, we would need to ensure that CRDs are installed before the Service is created. This is usually the case when CRDs are placed within the `crds` directory, but in this case, MetalLB [does not define them there](https://github.com/bitnami/charts/blob/main/bitnami/metallb/templates/controller/crds.yaml) (because they need to be templatized). 

This, combined with the fact that CRDs' registration process usually takes some time, makes using the `extraDeploy` an unsafe option: the K8s cluster cloud provider's native LB controller could interfere with MetalLB here.

Hence, I would keep this as it is to ensure the resource is deployed one MetalLB is marked as `ready` 😁